velox/connectors/Connector.h

```diff
@@ -144,10 +144,21 @@ class DataSink {
 
   /// Called once after all data has been added via possibly multiple calls to
   /// appendData(). Could return data in the string form that would be included
-  /// in the output. After calling this function, only close() could be called.
+  /// in the output.
+  ///
+  /// NOTE: we only expect finish() call on a closed data sink object.
   virtual std::vector<std::string> finish() const = 0;
 
+  /// TODO: consider to merge finish() into close() to returns the metadata in
```

Comment:
It seems confusing to have finish(), close() and abort(). It looks like on success, we expect close() followed by finish() and on failure we expect abort(). Also, not clear what finish() returns? What is in the vector<string>?

Would it make sense to have a single method that takes a boolean indicating whether we close on success or failure?

std::vector<std::string> close(bool success)

Reply:
Yes, that's what I thought and left a TODO there. Need to confirm with other Meta internal team to see if it matches their use case. Thanks!